src/GraphQL.Tests/Types/AllScalarGraphTypeTests.cs

```diff
@@ -0,0 +1,807 @@
+using System;
+using System.Numerics;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Types
+{
+    public class AllScalarGraphTypeTests
+    {
+        /*  This class tests the following scalar types:
+         *    ByteGraphType
+         *    SByteGraphType
+         *    ShortGraphType
+         *    UShortGraphType
+         *    IntGraphType
+         *    UIntGraphType
+         *    LongGraphType
+         *    ULongGraphType
+         *    
+         *    BooleanGraphType
+         *    FloatGraphType
+         *    StringGraphType
+         *    
+         *  Does not test:
+         *    IdGraphType
+         *    DecimalGraphType
+         *    UriGraphType
+         *    date/time graph types
+         *    enumeration graph types
+         *    
+         *  Does test ALL scalars' handling of null
+         *  
+         */
+
+        [Theory]
+        [InlineData(typeof(BooleanGraphType))]
+        [InlineData(typeof(ByteGraphType))]
+        [InlineData(typeof(SByteGraphType))]
+        [InlineData(typeof(ShortGraphType))]
+        [InlineData(typeof(UShortGraphType))]
+        [InlineData(typeof(IntGraphType))]
+        [InlineData(typeof(UIntGraphType))]
+        [InlineData(typeof(LongGraphType))]
+        [InlineData(typeof(ULongGraphType))]
+        [InlineData(typeof(BigIntGraphType))]
+        [InlineData(typeof(DateGraphType))]
+        [InlineData(typeof(DateTimeGraphType))]
+        [InlineData(typeof(DateTimeOffsetGraphType))]
+        [InlineData(typeof(TimeSpanSecondsGraphType))]
+        [InlineData(typeof(TimeSpanMillisecondsGraphType))]
+        [InlineData(typeof(IdGraphType))]
+        [InlineData(typeof(StringGraphType))]
+        [InlineData(typeof(UriGraphType))]
+        [InlineData(typeof(GuidGraphType))]
+        [InlineData(typeof(FloatGraphType))]
+        [InlineData(typeof(DecimalGraphType))]
+        [InlineData(typeof(EnumerationGraphType))]
+        public void allow_null(Type graphType)
+        {
+            var g = (ScalarGraphType)graphType.GetConstructor(Type.EmptyTypes).Invoke(null);
+            g.ParseValue(null).ShouldBeNull();
+            g.CanParseValue(null).ShouldBeTrue();
+            g.ParseLiteral(new NullValue()).ShouldBeNull();
```

Comment:
Somewhere here should be test for   `g.ParseLiteral(null)` and `g.CanParseLiteral(null)`

Reply:
Those values are undefined.

Reply:
Added test